### PR TITLE
Use `subrepo` to link to PyDev.Debugger instead of copying it

### PR DESCRIPTION
**Pull request by rchiodo**
This changes debugpy to [subrepo](https://github.com/ingydotnet/git-subrepo) PyDev.Debugger instead of just having a copy of it. This should make it easier to push commits between the two repositories.

Subrepo isn't like git submodule though. We are sync'd to a specific commit of PyDev.Debugger. Any changes to PyDev.Debugger have to be pulled into debugpy (and vice versa changes here have to be pushed to PyDev.Debugger).

I also made changes to get the latest commit PyDev.Debugger to work with debugpy. There was a problem with lambdas in pydevd_sys_monitoring messing up entry point breakpoints. The fix I made was to eliminate the lambda usage.

### CODE REVIEW

**Review by debonte — APPROVED** [review]

File: src/debugpy/_vendored/pydevd/pydevd_attach_to_process/windows/compile_windows.bat [hunk 1]
```diff
@@ -18,7 +18,7 @@ cl -DUNICODE -D_UNICODE /EHsc /Zi /O1 /W3 /LD /MD /D BITS_32 /Qspectre run_code_
 copy run_code_on_dllmain_x86.dll ..\run_code_on_dllmain_x86.dll /Y
 copy run_code_on_dllmain_x86.pdb ..\run_code_on_dllmain_x86.pdb /Y
 
-cl /EHsc /Zi /O1 /W3 /Qspectre inject_dll.cpp /link /PROFILE /GUARD:CF /out:inject_dll_x86.exe
+cl /EHsc /Zi /O1 /W3 /Qspectre inject_dll.cpp /link /PROFILE /GUARD:CF /CETCOMPAT /out:inject_dll_x86.exe
```
Comment (AdamYoblick) [inline src/debugpy/_vendored/pydevd/pydevd_attach_to_process/windows/compile_windows.bat:21]: Will we have to fix these compiler and linker flags up every time we pull from PyDev.Debugger, or are we trying to push these changes back to him?

Comment (rchiodo) [reply src/debugpy/_vendored/pydevd/pydevd_attach_to_process/windows/compile_windows.bat:21]: I'm going to push this back to him now

Comment (rchiodo) [reply src/debugpy/_vendored/pydevd/pydevd_attach_to_process/windows/compile_windows.bat:21]: I had made these changes there but missed a spot, this is my changes coming back

**Comment by rchiodo** [comment]
Here's the PR into PyDev.Debugger:

https://github.com/fabioz/PyDev.Debugger/pull/295

I made this with just the subrepo command:
`python subrepo.py branch -m "rchiodo/update_from_debugpy`

And then:
`git push -f https://github.com/rchiodo/PyDev.Debugger subrepo/src/debugpy/_vendored/pydevd:rchiodo/update_from_debugpy`

**Pull request merged**